src/components/Badge/Badge.tsx

```diff
@@ -0,0 +1,85 @@
+import classNames from "classnames";
+import React from "react";
+import type { ClassName, ValueOf } from "types";
+
+export const BadgeType = {
+  ROUNDED_LARGE_NUMBER: "ROUNDED_LARGE_NUMBER",
+  UNDEFINED_LARGE_NUMBER: "UNDEFINED_LARGE_NUMBER",
+} as const;
+
+/**
+ * The props for the Badge component.
+ */
+export type Props = {
+  /**
+   * Numeric value to be displayed.
+   */
+  value: number;
+  /**
+   * The type of the badge component.
+   */
+  badgeType?: ValueOf<typeof BadgeType> | string;
+  /**
+   * The appearance of the badge.
+   */
+  isNegative?: boolean;
```

Comment:
This name gets really confusing if you think about it in the context of numbers (and negative numbers)…

Reply:
Should we call it "isDark"?